layout/brandeskopf.go

```diff
@@ -158,12 +164,43 @@ func horizontalCompaction(g LayeredGraph, root map[uint64]uint64, align map[uint
 		}
 	}
 
+	// class offsets
+	layers := g.Layers()
+	for i := 0; i < len(layers); i++ {
+		layer := layers[i]
+		vfirst := layer[0]
+		if sink[vfirst] == vfirst {
+			if shift[sink[vfirst]] == math.MaxInt {
+				shift[sink[vfirst]] = 0
+			}
+			j := i
+			k := 0
```

Comment:
parallel assignment, also similar to paper notation
```suggestion
			j, k := i, 0
```